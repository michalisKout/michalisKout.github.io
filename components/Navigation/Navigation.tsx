import Toolbar from './components/Toolbar';
import NavItems from './components/NavItems';
import PrimaryLink from '../Shared/PrimaryLink';
import { useDarkModeClassName } from '../../core/hooks/utils/useDarkModeClassName';

const Navigation = () => {
  const darkModeClass = useDarkModeClassName();

  return (
    <div className={darkModeClass}>
      <nav className="nav">
        <div className="nav--container">
          <div className="nav--right">
            <PrimaryLink />
          </div>
          <div className="nav--left">
            <NavItems />
            <Toolbar />
          </div>
        </div>
      </nav>
    </div>
  );
};

export default Navigation;
