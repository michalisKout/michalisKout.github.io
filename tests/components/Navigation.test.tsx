import { render } from "@testing-library/react";
import Navigation from "../../components/Navigation/Navigation";

describe("Navigation Component", () => {
  beforeEach(() => {
    jest.mock("contentful", () => {
      return {
        creatClient: () => {},
      };
    });
  });

  it("should render Navigation", () => {
    const { asFragment } = render(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
