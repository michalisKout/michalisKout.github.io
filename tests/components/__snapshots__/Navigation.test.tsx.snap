// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should render Navigation 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <div
      class="nav--container"
    >
      <div
        class="nav--right"
      >
        <div>
          <a
            class="link-primary"
            href="#"
          >
            Brand
          </a>
        </div>
      </div>
      <div
        class="flex md:hidden"
      >
        <button
          aria-label="toggle menu"
          class="nav-burger-mobile"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-w-14 "
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="nav--left"
      >
        <div
          class="nav-items--wrapper"
        >
          <span
            class="nav-item"
            data-testid="nav-item"
          >
            <a
              href="/"
            >
              Home
            </a>
          </span>
          <span
            class="nav-item"
            data-testid="nav-item"
          >
            <a
              href="/portfolio"
            >
              Portfolio
            </a>
          </span>
          <span
            class="nav-item"
            data-testid="nav-item"
          >
            <a
              href="/experience"
            >
              Experience
            </a>
          </span>
          <span
            class="nav-item"
            data-testid="nav-item"
          >
            <a
              href="/contact"
            >
              Contact
            </a>
          </span>
        </div>
        <div
          class="toolbar"
        >
          <div
            class="relative"
          >
            <span
              class="search--container"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-search fa-w-16 "
                data-icon="search"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  fill="currentColor"
                />
              </svg>
            </span>
            <input
              class="search--input"
              placeholder="Search"
              type="text"
            />
          </div>
          <div
            class="switch"
            data-testid="switch-button"
          >
            <input
              readonly=""
              type="checkbox"
            />
            <span
              class="slider"
            />
          </div>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
